client/consensus/beefy/src/worker.rs

```diff
@@ -759,19 +765,29 @@ where
 		);
 
 		let message = VoteMessage { commitment, id: authority_id, signature };
-
-		let encoded_message = message.encode();
+		let gossip_vote = GossipMessage::<B>::Vote(message);
+		let encoded_vote = gossip_vote.encode();
+		let vote = if let GossipMessage::<B>::Vote(message) = gossip_vote {
```

Comment:
Could probable be moved couple of lines above and replaced with simpler construct?
```rust
let vote = message.clone();
```

Reply:
sure, I was trying to avoid the extra `.clone()` for every vote